source/logrepl/combined.go

```diff
@@ -111,27 +111,29 @@ func NewCombinedIterator(ctx context.Context, pool *pgxpool.Pool, conf Config) (
 	return c, nil
 }
 
-// Next provides the next available record from the snapshot or CDC stream.
-// If the end of the snapshot is reached, next will switch to the CDC iterator and retrive
-// the next available record. Failure to switch the iterator will return an error.
-func (c *CombinedIterator) Next(ctx context.Context) (opencdc.Record, error) {
-	r, err := c.activeIterator.Next(ctx)
+// NextN retrieves up to n records from the active iterator.
+// If the end of the snapshot is reached during this call, it will switch to the CDC iterator
+// and continue retrieving records from there.
+func (c *CombinedIterator) NextN(ctx context.Context, n int) ([]opencdc.Record, error) {
+	if n <= 0 {
+		return nil, fmt.Errorf("n must be greater than 0, got %d", n)
+	}
+
+	records, err := c.activeIterator.NextN(ctx, n)
 	if err != nil {
 		// Snapshot iterator is done, handover to CDC iterator
-		if !errors.Is(err, snapshot.ErrIteratorDone) {
-			return opencdc.Record{}, fmt.Errorf("failed to fetch next record: %w", err)
-		}
-
-		if err := c.useCDCIterator(ctx); err != nil {
-			return opencdc.Record{}, err
+		if errors.Is(err, snapshot.ErrIteratorDone) {
```

Comment:
Nitpick, but why increase the nesting here?

Reply:
@lyuboxa i had actually changed how these errors operated by switching to CDC explicitly calling to `.NextN(` [later on](https://github.com/ConduitIO/conduit-connector-postgres/pull/276/files?diff=unified&w=0#diff-16bc8373489d2c228c26cf08f43674185eacb1d97804697ef714742a5335238aR132). I realize that maybe it's better actually switch to CDC when the snapshot iterator finishes, so I'll change this back a bit.

Reply:
@lyuboxa https://github.com/ConduitIO/conduit-connector-postgres/pull/276/commits/89f2f0caeb188f4cc6c3d73e307ff75eb2eb7d59